src/main/java/com/redhat/devtools/lsp4ij/features/documentation/LSPDocumentationHelper.java

```diff
@@ -26,63 +30,115 @@ private LSPDocumentationHelper() {
-            return htmlBody;
+    public static @NotNull List<MarkupContent> getValidMarkupContents(@Nullable Hover hover) {
+        if (hover == null) {
+            // textDocument/hover may return null
+            return Collections.emptyList();
         }
-        Color background = editor != null ? editor.getColorsScheme().getDefaultBackground() : null;
-        Color foreground = editor != null ? editor.getColorsScheme().getDefaultForeground() : null;
-
-        StringBuilder html = new StringBuilder("<html><head><style TYPE='text/css'>html { ");
-        if (background != null) {
-            html.append("background-color: ")
-                    .append(toHTML(background))
-                    .append(";");
+        var hoverContents = hover.getContents();
+        if (hoverContents == null) {
+            return Collections.emptyList();
         }
-        if (foreground != null) {
-            html.append("color: ")
-                    .append(toHTML(foreground))
-                    .append(";");
+        if (hoverContents.isLeft()) {
+            var contents = hoverContents.getLeft();
+            if (contents == null || contents.isEmpty()) {
+                return Collections.emptyList();
+            }
+            return contents.stream()
+                    .map(content -> {
+                        if (content.isLeft()) {
+                            String value = content.getLeft();
+                            if (!isValidContent(value)) {
+                                return null;
+                            }
+                            return new MarkupContent(MarkupKind.MARKDOWN, value);
+                        } else if (content.isRight()) {
+                            MarkedString markedString = content.getRight();
+                            if (!isValidContent(markedString)) {
+                                return null;
+                            }
+                            String value = markedString.getValue();
+                            String language = markedString.getLanguage() != null && !markedString.getLanguage().isEmpty() ?
+                                    markedString.getLanguage() :
+                                    null;
+                            if (language == null) {
+                                return new MarkupContent(MarkupKind.MARKDOWN, value);
+                            }
+                            value = String.format("```%s%n%s%n```", language, value);
+                            return new MarkupContent(MarkupKind.MARKDOWN, value);
+                        }
+                        return null;
+                    })
+                    .filter(Objects::nonNull)
+                    .toList();
+        } else if (hoverContents.isRight()){
+            MarkupContent content = hoverContents.getRight();
+            if (!isValidContent(content)) {
+                return Collections.emptyList();
+            }
+            return List.of(content);

```

Comment:
replace with

```java
            if (isValidContent(content)) {
                return List.of(content);
            }
```
it'll fall back to the last return statement if not valid

Reply:
fixed